### PR TITLE
 #166241006 - Listing the Users profile

### PR DESCRIPTION
**Pull request by UhiriweAudace**
#### What does this PR do?
In this PR, I implemented the endpoints that help any user/client to be able to view a list of all user's profile
#### Description of Task to be completed?
Having the following Endpoints working:
`GET /api/profiles`
#### How should this be manually tested?
- clone the repository
- go to your terminal and then write the following command
  `cd tesla-ah`
- after that try to install all dependencies that are required for this project, simply by running the following command:
   `yarn install`  or `yarn`
- Then running the server, by running the below commands:
   1. `sequelize db:migrate` for this command make sure that the `sequelize-cli` package is installed globally
   2. `yarn dev`
   3.  After running the server, Open your postman, and then try to send a request of getting user profiles
- for running the tests, write the below command in your terminal
     `yarn test:local`
 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166241006 ](https://www.pivotaltracker.com/n/projects/2350798/stories/166241006)
#### Screenshots (if appropriate)
N/A
#### Questions:

****
**Before Code - Audace Uhiriwe**
https://docs.google.com/document/d/1bWYaHMgb3SyKN9QfmD9KbeX-WeDGHjplg66l8wOdYso/edit?usp=sharing

### CODE REVIEW

File: src/api/routes/userRouter.js [hunk 1]
```diff
@@ -2,11 +2,12 @@ import { Router } from 'express';
...
 import Auth from '../../middleware/auth';
 import validateBody from '../../middleware/validateBody';
+import check from '../../middleware/checkOwner';
 
 const userRouter = Router();
 const { updateProfile } = ProfilesController;
 const { verifyToken } = Auth;
 
-userRouter.put('/', verifyToken, validateBody('updateUser'), updateProfile);
+userRouter.put('/', verifyToken, check.profileOwner, validateBody('updateUser'), updateProfile);
```
Comment (deschantkn) [inline src/api/routes/userRouter.js]: This middleware is not necessary as token verification ensures the user exists. What can be done is to update the `404` status code in the controller to `403`.

File: src/middleware/checkOwner.js [hunk 2]
```diff
@@ -31,5 +31,19 @@ class checkOwner {
...
   }
+
+  /**
+   * @author Audace Uhiriwe
+   * @param {req} - request
+   * @param {res} - response
+   */
+  static async profileOwner(req, res, next) {
+    const { id, username } = req.user;
+    const userFound = await User.findOne({ where: { id, username } });
```
Comment (deschantkn) [inline src/middleware/checkOwner.js]: How does this verify profile ownership? It is just repeating what token verification does. You should remove it.

**Review by deschantkn — DISMISSED** [review]

**Review by eliemugenzi — DISMISSED** [review]

File: src/api/controllers/profiles.js [hunk 3]
```diff
@@ -31,18 +31,28 @@ export default class ProfilesController {
...
+
+  /**
+   * @Author - Audace Uhiriwe
+   * @param  {object} req
+   * @param  {object} res
+   * @return {object} returns an object containing the user's profiles
+   */
+  static async getAllProfile(req, res) {
+    const fetchedProfile = await User.findAll({ attributes: ['username', 'firstName', 'lastName', 'bio', 'image'] });
+    if (!fetchedProfile[0]) return res.status(404).send({ error: 'No Users Profiles found!' });
```
Comment (eliemugenzi) [inline src/api/controllers/profiles.js]: This is not a client error, it should not be 404.

**Review by eliemugenzi — APPROVED** [review]

**Review by EmyRukundo — APPROVED** [review]

**Review by Quantum-35 — APPROVED** [review]

File: src/api/controllers/profiles.js (hunk 3 above)
Comment (UhiriweAudace) [reply src/api/controllers/profiles.js]: @eliemugenzi Thanks! i'm going to fix it

**Pull request merged**